add `--js` alias to `--javascript` and `--skip-js` alias to `--skip-javscript`

dorianmariecom opened the pull request:
### Summary

I was creating a new app and did `rails new --js esbuild` thinking it would work, but it didn't and was surprised, so here I add the `--js` alias so that `rails new --javascript ...` has two aliases: `-j` and `--js`

### Other Information

I'm not too familiar with thor so not sure 100% especially about the change in `lib/rails/generators.rb`, not sure if it has any effect at all, doesn't seem to break tests though

Because it's using thor, the `rails new --help` is updated automatically:

```
  -j, --js, [--javascript=JAVASCRIPT]                            # Choose JavaScript approach [options: importmap (default), webpack, esbuild, rollup]
```

In https://github.com/rails/rails/pull/43563 @eileencodes suggested I do this instead of renaming `--javascript` to `--js`

**Edit**: Also did the same for `--skip-js` / `--skip-javascript`

dorianmariefr commented:
Should I rebase?

byroot reviewed (approved):
If you rebase I'll happily merge.

While you are at it, please merge the two CHANGELOG entries together.

Commit:
Add `--js` and --skip-javascript` options to `rails new`

`--js` alias to `rails new --javascript ...`

Same as `-j`, e.g. `rails new --js esbuild ...`

`--skip-js` alias to `rails new --skip-javascript ...`

Same as `-J`, e.g. `rails new --skip-js ...`

dorianmariefr commented:
thanks @byroot 🙏
